lightning/src/routing/router.rs

```diff
@@ -1228,7 +1224,7 @@ fn iter_equal<I1: Iterator, I2: Iterator>(mut iter_a: I1, mut iter_b: I2)
 pub struct PathBuildingHop<'a> {
 	// Note that this should be dropped in favor of loading it from CandidateRouteHop, but doing so
 	// is a larger refactor and will require careful performance analysis.
-	node_id: NodeId,
```

Comment:
Some places you replace uses of `node_id` with `candidate.source()` and other places with `candidate.target()`. How can it be both?